frontend/src/queries/nodes/InsightViz/InsightViz.tsx

```diff
@@ -70,7 +70,9 @@ export function InsightViz({ query, setQuery }: InsightVizProps): JSX.Element {
                     'insight-wrapper--singlecolumn': isFunnels,
                 })}
             >
-                <EditorFilters query={query.source} setQuery={setQuerySource} />
+                {!!insight.query && isInsightVizNode(insight.query) && (
```

Comment:
I'm letting any query into this view now. So only want the editor filters if it's an insight vizualisation...

> In general I feel like many logics are too big and I'd like to separate them out to "do one thing well".

Totally agree with this...

I'll make it so the `insightDataLogic` ignores non graphing `Node`s rather than start shoving other types into it too

Reply:
I'm not sure I understand what's happening yet. Disambiguation for the queries should happen in `frontend/src/queries/Query/Query.tsx`, or is there a deeper reason on why we'd want other queries to go to `InsightsViz`?

Reply:
I was following the types and got turned around. This doesn't need to go into InsightVizNode and doesn't any more 😊